srcts/src/shiny/shinyapp.ts

```diff
@@ -633,19 +633,19 @@ class ShinyApp {
 
   // A function for sending messages to the appropriate handlers.
   // - msgObj: the object containing messages, with format {msgObj.foo, msObj.bar
-  private _sendMessagesToHandlers(
+  private async _sendMessagesToHandlers(
     msgObj: { [key: string]: unknown },
     handlers: { [key: string]: Handler },
     handlerOrder: string[]
-  ): void {
+  ): Promise<void> {
     // Dispatch messages to handlers, if handler is present
     for (let i = 0; i < handlerOrder.length; i++) {
       const msgType = handlerOrder[i];
 
       if (hasOwnProperty(msgObj, msgType)) {
         // Execute each handler with 'this' referring to the present value of
         // 'this'
-        handlers[msgType].call(this, msgObj[msgType]);
+        await handlers[msgType].call(this, msgObj[msgType]);
```

Comment:
This now allows for message handlers to be `async`/`await`-able

Reply:
This spooks me as we are adding the possibility that the handling of messages can occur in multiple steps, allowing other events to happen in between processing. 

Before, the _handshake_ was that all messages were handling in a single tick. Now, the execution could take many ticks and introduce many edge cases that are hard to reprex.

Reply:
The `async`-ness is unfortunately necessary because `renderDependency` needs to be `async`, and that has implications all the way up the chain.

Adding the `await` here doesn't actually change whether or not a handler can be `async`. Even prior to this change, they could be `async`; it's just that they would run, create a promise, and then the code would move on without waiting for the promise to resolve.